README.rst: create by pandoc rather than pypandoc

ickc opened the pull request:
1. travis can test the validity of the README.rst.
    - It is for PyPI’s README, since an invalid rst will be shown as plain text over PyPI.
    - a failure on rst2html conversion will *not* fail the whole build. But it will be useful to check for errors in Travis when one see the PyPI's README is plaintext.
2. It also removes the extra dep on pypandoc

sergiocorreia commented:
I'm getting build failures related to this:

$ rst2html.py README.rst > README.html
/home/travis/build.sh: line 57: rst2html.py: command not found
The command "rst2html.py README.rst > README.html" failed and exited with 127 during .

Do you know what might be going on? (is rst2html.py not always distributed with python?)

ickc commented:
I forgot to add docsutils to the dependencies. It should be fixed in #61 now.

More elaboration on the rst2html in travis:

I added the rst2html.py because I often find PyPI will shows the README as plain text rather than a rendered (pandoc md to rst, then rst to html) output: e.g. <https://pypi.python.org/pypi/panflute/1.10.1>.

So when one see the PyPI README is plain text only, one can now go to the travis and read the deploy msg (look for rst2html) and see what warning it is emitting (e.g. in past experience, if I syntax highlight markdown by `~~~markdown`, PyPI don't like it because markdown is not a valid language).

But since that piece of code is in Travis' `before_deploy`, it wasn't tested in my Travis (since I'm not deploying).

So I temporarily put that piece of code output `before_deploy` and tested it in Travis: [Job #151.8 - ickc/panflute - Travis CI](https://travis-ci.org/ickc/panflute/jobs/192503595), and it is working. (I reverted that temp change, so it is now still in `before_deploy`.)

ickc commented:
By the way, even if rst2html.py emits warning, **it will not fails the build**. It is good because PyPI's README is relatively unimportant and should not be blocking for a deployment. And those warnings are there for reference only.
